media_driver/linux/common/codec/ddi/media_ddi_decode_vp8.cpp

```diff
@@ -84,12 +84,17 @@ VAStatus DdiDecodeVP8::ParseSliceParams(
     picParams->CodedCoeffTokenPartition = (num_token_partitions != 8) ? (num_token_partitions >> 1) : 3;
     //macroblock_offset is in unit of bit.it should be always the next byte, the byte is divided to two parts
     //used bits and remaining bits, if used bits == 8, uiFirstMbByteOffset should add 1, so use 8 to do the ceil operator
-    picParams->uiFirstMbByteOffset = slcParam->slice_data_offset + ((slcParam->macroblock_offset + 8) >> 3);
+    picParams->uiFirstMbByteOffset = slcParam->slice_data_offset + ((slcParam->macroblock_offset + 7) >> 3);
 
     memcpy_s(picParams->uiPartitionSize, sizeof(picParams->uiPartitionSize), slcParam->partition_size, sizeof(picParams->uiPartitionSize));
 
     //partition 0 size in command buffer includes the one byte in bool decoder if remaining bits of bool decoder is not zero.
-    picParams->uiPartitionSize[0] += (slcParam->macroblock_offset & 0x7) ? 1 : 0;
+    //bool coder count should be 0-7, if it is 0 (ucP0EntropyCount == 0), actually it means there are 8 bit is loaded in bool coder.
+    if(picParams->ucP0EntropyCount == 0)
```

Comment:
Doubt it works. Because after Vp8EntropyState::DecodeFill() m_count is between 16 and 24. After some decoding it is 0<=m_count<24. Then in ParseFrameHead() I see vp8PicParams->ucP0EntropyCount = 8 - (m_count & 0x07); here byte-offset part of m_count is lost. The range is [1;8].
Then here in 'if (picParams->ucP0EntropyCount == 0)' byte offset is partially lost and the condition can never be true.

Reply:
Alternative is
https://github.com/Intel-Media-SDK/MediaSDK/pull/493
https://github.com/lakulako/media-driver/pull/1

Reply:
mostlikely , we should calculate the value and bool coder remaining bits together